src/util/adb.js

```diff
@@ -214,6 +214,41 @@ export default class ADBUtils {
     return artifactsDir;
   }
 
+  async checkOrCleanArtifacts(
+    deviceId: string, clean: boolean): Promise<boolean> {
+    const {adbClient} = this;
+    const directories = [];
+    await wrapADBCall(async () => {
+      log.debug('Finding Older Artifacts');
+      const dirs = await adbClient.readdir(deviceId, '/sdcard');
+      for (const dir of dirs) {
+        if (!dir.isFile()) {
+          if (dir.name.startsWith('web-ext-artifacts-')) {
+            directories.push(dir.name);
+          }
+        }
+      }
+      if (clean) {
+        for (var artifactsDir of directories) {
+          const temp = '/sdcard/';
+          artifactsDir = temp + artifactsDir;
+          await this.runShellCommand(deviceId, [
+            'rm', '-rf', artifactsDir,
+          ]);
+        }
+      }
+    });
+    if (directories.length !== 0) {
+      //If directories found and condition is to clean them
+      if (clean) {
+        return false;
+      }
+      //If directories found and condition is to only check
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
it looks that this method can be made a bit simpler.

e.g. (as I mentioned in my previous comment) it doesn't seem that we really need to collect the directories into an array and then iterate again over it.

Besides that, all these if/else blocks between line 241-250 could be replaced with something a bit cleaner (and less nested).

Try to ask yourself some questions to help you get a better picture of what are options you may have here, e.g.

- do we always need to iterate over all the results resolved by `adb.readdir` or could we exit earlier in certain cases? (e.g. if a call to this method is meant to only check if any old artifact dir is still on the device, in which conditions this would be are ready to return to the caller the outcome of that check?)